src/TConsole.cpp

```diff
@@ -2949,3 +2950,41 @@ void TConsole::setProfileName(const QString& newName)
         pC->setProfileName(newName);
     }
 }
+
+
+void TConsole::dragEnterEvent(QDragEnterEvent* e)
+{
+    if (e->mimeData()->hasUrls()) {
+        e->acceptProposedAction();
+    }
+}
+
+//https://amin-ahmadi.com/2016/01/04/qt-drag-drop-files-images/
+void TConsole::dropEvent(QDropEvent* e)
+{
+    QStringList accepted_types;
+    accepted_types << "jpeg"
+                   << "jpg"
+                   << "png";
+    foreach(const QUrl& url, e->mimeData()->urls()) {
+        QString fname = url.toLocalFile();
+        QFileInfo info(fname);
+        if (info.exists()) {
+            if (accepted_types.contains(info.suffix().trimmed(), Qt::CaseInsensitive)) {
+                QPoint pos = e->pos();
+                TEvent mudletEvent{};
+                mudletEvent.mArgumentList.append(QLatin1String("sysImageDropEvent"));
+                mudletEvent.mArgumentList.append(fname);
+                mudletEvent.mArgumentList.append(QString::number(pos.x()));
+                mudletEvent.mArgumentList.append(QString::number(pos.y()));
+                mudletEvent.mArgumentList.append(mConsoleName);
+                mudletEvent.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+                mudletEvent.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+                mudletEvent.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+                mudletEvent.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+                mudletEvent.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
```

Comment:
For scripters convenience - I think also giving them the file extension would be nice, so everybody doesn't have to duplicate trimming the filename.

Reply:
You are right. Good idea.
Will change it to just sysDropEvent and allow to drop every kind of file and then with the given file extension the scripters should know what to do.
Like this it will also be possible to drag and drop an mpackage/xml and automatically install it.

Reply:
Yep... and we should ship that by default. Always wanted it for a while.